### PR TITLE
Further edits to the JS chapter

### PR DESCRIPTION
**Pull request by tkadlec**
These edits should resolve the open comments in #1737.

I also added a quick blurb on module/nomodule since I thought I already had something in place there, but apparently not. :/

Progress on #897 #1432

### CODE REVIEW

File: src/content/en/2020/javascript.md [hunk 1]
```diff
@@ -112,6 +111,13 @@ As raw numbers, those may or may not jump out at you depending on how much of a
 
 That 153 KB equates to ~37% of the total script size that we send down to mobile devices. There's definitely some room for improvement here.
 
+### Module/NoModule
```
Comment (rviscomi) [inline src/content/en/2020/javascript.md:114]:
```suggestion
### `module` and `nomodule`
```

File: src/content/en/2020/javascript.md [hunk 2]
```diff
@@ -112,6 +111,13 @@ As raw numbers, those may or may not jump out at you depending on how much of a
...
+One mechanism we have to potentially reduce the amount of code we send down is to take advantage of the [module/nomodule pattern](https://web.dev/serve-modern-code-to-modern-browsers/). With this pattern, we create two sets of bundles: one bundle intended for modern browsers and one intended for legacy browsers. The bundle intended for modern browsers gets a `type=mobdule` and the bundle intended for legacy browsers gets a `type=nomodule`.
```
Comment (rviscomi) [inline src/content/en/2020/javascript.md:115]:
```suggestion
One mechanism we have to potentially reduce the amount of code we send down is to take advantage of the [`module`/`nomodule` pattern](https://web.dev/serve-modern-code-to-modern-browsers/). With this pattern, we create two sets of bundles: one bundle intended for modern browsers and one intended for legacy browsers. The bundle intended for modern browsers gets a `type=module` and the bundle intended for legacy browsers gets a `type=nomodule`.
```

File: src/content/en/2020/javascript.md [hunk 3]
```diff
@@ -112,6 +111,13 @@ As raw numbers, those may or may not jump out at you depending on how much of a
...
+
+This approach lets us create smaller bundles with optimized modern syntax for the browsers that support it, while providing conditionally loaded polyfills and different syntax to the browsers that don't.
```
Comment (rviscomi) [inline src/content/en/2020/javascript.md:117]:
```suggestion
This approach lets us create smaller bundles with modern syntax optimized for the browsers that support it, while providing conditionally loaded polyfills and different syntax to the browsers that don't.
```

File: src/content/en/2020/javascript.md [hunk 4]
```diff
@@ -112,6 +111,13 @@ As raw numbers, those may or may not jump out at you depending on how much of a
...
+
+Support for `module` and `nomodule` is broadening, but still relatively new. As a result, adoption is still a bit low. Only 3.6% of mobile pages use at least one script with `type=module` and only .7% of mobile pages use at least one script with `type=nomodule` to support legacy browsers.
```
Comment (rviscomi) [inline src/content/en/2020/javascript.md:119]:
```suggestion
Support for `module` and `nomodule` is broadening, but still relatively new. As a result, adoption is still a bit low. Only 3.6% of mobile pages use at least one script with `type=module` and only 0.7% of mobile pages use at least one script with `type=nomodule` to support legacy browsers.
```

File: src/content/en/2020/javascript.md [hunk 5]
```diff
@@ -217,13 +223,17 @@ Of those external scripts, only 12.2% of them are loaded with the `async` attrib
...
+Nowadays, the pattern is unnecessary for the vast majority of sites and any script loaded with the pattern in place will interrupt the HTML parser when it needs to be executed, instead of deferring until the page has loaded. The pattern is still used surprisingly often, with 11.4% of mobile pages serving at least one script with that pattern in place. In other words, at least some of the 6% of scripts that use `defer` aren't getting the full benefits of the `defer` attribute.
+
+There is an encouraging story here, though.
+
+Harry Roberts [tweeted about the antipattern on Twitter](https://twitter.com/csswizardry/status/1331721659498319873), which is what prompted us to check to see how frequently this was occurring in the wild. [Rick Viscomi checked to see who the top culprits were](https://twitter.com/rick_viscomi/status/1331735748060524551), and it turns out "stats.wp.com" was the source of the most common offenders. Someone from Automatic replied and the pattern will now be [removed going forward](https://twitter.com/Kraft/status/1336772912414601224).
```
Comment (rviscomi) [inline src/content/en/2020/javascript.md:234]:
```suggestion
Harry Roberts [tweeted about the antipattern on Twitter](https://twitter.com/csswizardry/status/1331721659498319873), which is what prompted us to check to see how frequently this was occurring in the wild. [Rick Viscomi checked to see who the top culprits were](https://twitter.com/rick_viscomi/status/1331735748060524551), and it turns out "stats.wp.com" was the source of the most common offenders. @Kraft from Automattic replied, and the pattern will now be [removed going forward](https://twitter.com/Kraft/status/1336772912414601224). 
```

File: src/content/en/2020/javascript.md [hunk 6]
```diff
@@ -630,8 +639,7 @@ As you would expect, there's a gap for all tools in use due to the lower process
...
 
-{# TODO(authors): Was there something you wanted to add at the end of this paragraph? "While browsers are increasingly finding ways to offload"... #}
-The first thing we should consider is what happens with all that JavaScript once its been downloaded. Downloading is only the first part of the JavaScript journey. The browser still has to parse all that script, compile it, and eventually execute it.
+The first thing we should consider is what happens with all that JavaScript once its been downloaded. Downloading is only the first part of the JavaScript journey. The browser still has to parse all that script, compile it, and eventually execute it. While browsers are constantly on the lookout for ways to offload some of that cost to other threads, much of that work still happens on the main thread, blocking the browser from being able to do layout or paint related work, as well as from being able to respond to user interaction.
```
Comment (rviscomi) [inline src/content/en/2020/javascript.md:642]:
```suggestion
The first thing we should consider is what happens with all that JavaScript once its been downloaded. Downloading is only the first part of the JavaScript journey. The browser still has to parse all that script, compile it, and eventually execute it. While browsers are constantly on the lookout for ways to offload some of that cost to other threads, much of that work still happens on the main thread, blocking the browser from being able to do layout or paint-related work, as well as from being able to respond to user interaction.
```

**Review by rviscomi — CHANGES_REQUESTED** [review]
LGTM just a few suggestions then this is good to go and we can remove the `unedited` flag. Thanks @tkadlec!

**Comment by tkadlec** [comment]
@rviscomi Thanks! Approved and committed them all.

**Review by rviscomi — APPROVED** [review]

**Pull request merged**